genesyscloud/user_roles/genesyscloud_user_roles_proxy.go

```diff
@@ -70,6 +72,11 @@ func getUserRolesByIdFn(_ context.Context, p *userRolesProxy, roleId string) (*[
 func updateUserRolesFn(_ context.Context, p *userRolesProxy, roleId string, rolesConfig *schema.Set, subjectType string) (*platformclientv2.APIResponse, error) {
 	// Get existing roles/divisions
 	subject, resp, err := p.authorizationApi.GetAuthorizationSubject(roleId, true)
+
+	if err != nil || resp.StatusCode == http.StatusNotFound || subject == nil {
```

Comment:
There was not a nil check here. So if the user entered a role name that was not real, the subject would not be retrieved and the code would fail with a nil pointer